internal/config/identity/openid/jwt.go

```diff
@@ -72,6 +77,10 @@ func (r *Config) PopulatePublicKey(arn arn.ARN) error {
 	if pCfg.JWKS.URL == nil || pCfg.JWKS.URL.String() == "" {
 		return nil
 	}
+
+	// Add client secret for the client ID for HMAC based signature.
+	r.pubKeys.add(pCfg.ClientID, []byte(pCfg.ClientSecret))
```

Comment:
NOTE: OpenID spec says HMAC keys will have `kid` that is equal to 'ClientID' and the signature is calculated via ClientSecret.